src/psyclone/tests/gocean/transformations/globalstoargs_test.py

```diff
@@ -0,0 +1,260 @@
+                           join(BASEPATH, "gocean1p0",
+                                "single_invoke_three_kernels_with_use.f90"),
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    trans = KernelGlobalsToArguments()
+
+    expected = [
+        ["subroutine kernel_with_use_code(ji,jj,istep,ssha,tmask,rdt)",
+         "real, intent(inout) :: rdt"],
+        ["subroutine kernel_with_use2_code(ji,jj,istep,ssha,tmask,cbfr,rdt)",
+         "real, intent(inout) :: cbfr\n  real, intent(inout) :: rdt"],
+        ["subroutine kernel_with_use_code(ji,jj,istep,ssha,tmask,rdt)",
+         "real, intent(inout) :: rdt"]]
+
+    for num, kernel in enumerate(invoke.schedule.coded_kernels()):
+        kschedule = kernel.get_kernel_schedule()
+
+        # We already tested that DEFERRED globals are resolved in last test,
+        # in this case we hardcode all them to REAL to avoid searching and
+        # importing of modules in this test.
+        for var in kschedule.symbol_table.global_datasymbols:
+            var._datatype = DataType.REAL
+
+        trans.apply(kernel)
+
+        # Check the kernel code is generated as expected
+        kernel_code = fwriter(kschedule)
+        assert expected[num][0] in kernel_code
+        assert expected[num][1] in kernel_code
+
+    generated_code = str(psy.gen)
+
+    # The following assert checks that globals from the same module are
+    # imported in a single use statement and that duplicated globals are
+    # just imported once
+    assert "SUBROUTINE invoke_0(oldu_fld, cu_fld)\n" \
+           "      USE kernel_with_use2_mod, ONLY: kernel_with_use2_code\n" \
+           "      USE kernel_with_use_mod, ONLY: kernel_with_use_code\n" \
+           "      USE model_mod, ONLY: rdt, cbfr\n" \
+           "      TYPE(r2d_field), intent(inout) :: cu_fld\n" in generated_code
+
+    # Check the kernel calls have the global passed as last argument
+    assert "CALL kernel_with_use_code(i, j, oldu_fld, cu_fld%data, " \
+           "cu_fld%grid%tmask, rdt)" in generated_code
+    assert "CALL kernel_with_use2_code(i, j, oldu_fld, cu_fld%data, " \
+           "cu_fld%grid%tmask, cbfr, rdt)" in generated_code
+
+
+def test_globalstoarguments_noglobals():
+    ''' Check the KernelGlobalsToArguments transformation can be applied to
+    a kernel that does not contain any global without any effect '''
+    from psyclone.transformations import KernelGlobalsToArguments
+
+    # Parse a file to get an initialised GOKernelsArguments object
+    _, invoke_info = parse(os.path.join(BASEPATH, "gocean1p0",
+                                        "single_invoke.f90"),
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    kernel = invoke.schedule.coded_kernels()[0]
+    trans = KernelGlobalsToArguments()
+    before_code = str(psy.gen)
+    trans.apply(kernel)
```

Comment:
Please could you add a TODO here referencing #11 as we may want to warn the user that no globals were found. In fact, could you add it in the `apply()` method too?

Reply:
Done